backend/de.metas.business/src/main/java/de/metas/order/paymentschedule/OrderPayScheduleRepository.java

```diff
@@ -0,0 +1,138 @@
+import de.metas.payment.paymentterm.PaymentTermId;
+import de.metas.payment.paymentterm.ReferenceDateType;
+import de.metas.util.Services;
+import de.metas.util.lang.Percent;
+import lombok.NonNull;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.compiere.model.I_C_Order;
+import org.compiere.model.I_C_OrderPaySchedule;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Repository
+public class OrderPayScheduleRepository
+{
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+	private final IOrderBL orderBL = Services.get(IOrderBL.class);
+
+	@NonNull
+	public I_C_OrderPaySchedule save(@NonNull final OrderPaySchedule schedule)
+	{
+		final I_C_OrderPaySchedule record = toRecord(schedule);
+		save(record);
+		return record;
+	}
+
+	public void save(@NonNull final I_C_OrderPaySchedule record)
+	{
+		InterfaceWrapperHelper.save(record);
+	}
+
+	@NonNull
+	public OrderPaySchedule getById(@NonNull final OrderPayScheduleId id)
+	{
+		final I_C_OrderPaySchedule record = InterfaceWrapperHelper.load(id, I_C_OrderPaySchedule.class);
+		return fromRecord(record);
+
+	}
+
+	@NonNull
+	public List<OrderPaySchedule> getByOrderId(@NonNull final OrderId orderId)
+	{
+		return queryBL
+				.createQueryBuilder(I_C_OrderPaySchedule.class)
+				.addEqualsFilter(I_C_OrderPaySchedule.COLUMNNAME_C_Order_ID, orderId)
+				.orderBy(I_C_OrderPaySchedule.COLUMNNAME_SeqNo)
+				.create()
+				.stream()
+				.map(this::fromRecord)
+				.collect(Collectors.toList());
+
+	}
+
+	public void deleteByOrderId(@NonNull final OrderId orderId)
+	{
+		queryBL
+				.createQueryBuilder(I_C_OrderPaySchedule.class)
+				.addEqualsFilter(I_C_OrderPaySchedule.COLUMNNAME_C_Order_ID, orderId)
+				.create()
+				.delete();
+
+	}
+
+	private I_C_OrderPaySchedule toRecord(@NonNull final OrderPaySchedule schedule)
+	{
+		final I_C_OrderPaySchedule record;
+		if (schedule.getId() != null)
+		{
+			record = InterfaceWrapperHelper.load(schedule.getId(), I_C_OrderPaySchedule.class);
+		}
+		else
+		{
+			record = InterfaceWrapperHelper.newInstance(I_C_OrderPaySchedule.class);
+		}
+
+		record.setC_Order_ID(schedule.getOrderId().getRepoId());
+		record.setC_PaymentTerm_Break_ID(schedule.getPaymentTermBreakId().getRepoId());
+		record.setDueAmt(schedule.getDueAmount().getAsBigDecimal());
+		record.setDueDate(schedule.getDueDate());
+		record.setPercent(schedule.getPercent().toInt());
+		record.setReferenceDateType(schedule.getReferenceDateType().getCode());
+		record.setStatus(OrderPayScheduleStatus.toCodeOrNull(schedule.getOrderPayScheduleStatus()));
+
+		return record;
+	}
+
+	private OrderPaySchedule fromRecord(@NonNull final I_C_OrderPaySchedule record)
+	{
+		final I_C_Order order = orderBL.getById(OrderId.ofRepoId(record.getC_Order_ID()));
+		final PaymentTermId paymentTermId = orderBL.getPaymentTermId(order);
```

Comment:
consider either
1. introduce C_OrderPaySchedule.C_PaymentTerm_ID (and ofc keep it in sync)
2. remove PaymentTermId from PaymentTermBreakId.